src/v/iceberg/transform_utils.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "iceberg/transform_utils.h"
+
+#include "iceberg/datatypes.h"
+#include "iceberg/transform.h"
+
+namespace iceberg {
+
+namespace {
+
+int32_t micros_to_hr(int64_t micros) {
+    static constexpr int64_t s_per_hr = 3600;
+    static constexpr int64_t micros_per_s = 1000000;
+    static constexpr int64_t micros_per_hr = micros_per_s * s_per_hr;
+    return static_cast<int32_t>(micros / micros_per_hr);
+}
+
+struct hourly_visitor {
```

Comment:
kinda debatable, but depending on how you think the set of transform utilities will grow it might be useful to consider having like "time_transform" or something like that otherwise transform_utils.cc will be a catch-all. nbd if its all expected to stay small

Reply:
Fair point, the rust impl also splits this out. I've moved the transform visitor into its own cc file